### DISCUSSION TITLE
How can I view users with write permissions to repo through the github api?

### DISCUSSION THREAD

**Discussion by chdsbd** [community/community#24875] [Apps, API and Webhooks]
<p>I’ve tried using the collaborators attribute from the graphql api, but that only returns users, without their permissions.</p>
<p>The v3 api seemed like it would work with <a href="https://developer.github.com/v3/repos/collaborators/#list-collaborators" rel="nofollow noopener">https://developer.github.com/v3/repos/collaborators/#list-collaborators</a>, but this api call doesn’t appear to return all the users with write access to a repo. In my case, I’m a member of a team on the pytest-dev org, which has write permission to the pytest repo, but when I query <a href="https://api.github.com/repos/pytest-dev/pytest/collaborators" rel="nofollow noopener">https://api.github.com/repos/pytest-dev/pytest/collaborators</a>, I do not see my user account.</p>
<p>How can I find users with write permissions to a repo through the github API? I need this to determine if a user’s review of a PR is meaningful, as only reviews from users with write permissions count.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Are you <a href="https://developer.github.com/v3/repos/collaborators/#list-collaborators" rel="nofollow noopener">passing the <code>hellcat-preview</code> media type as described in the documentation.</a>? I believe that is required to get the list of absolutely all collaborators, including ones on child teams.</p>
<p>I hope that helps!</p>
